pubsub/google/cloud/pubsub_v1/publisher/_batch/thread.py

```diff
@@ -137,6 +137,17 @@ def status(self):
         """
         return self._status
 
+    def wait(self):
+        """If commit in progress, waits until all of the futures resolved.
+
+        .. note::
+
+            This method blocks until all futures of this batch resolved.
+        """
+        if self._status in (base.BatchStatus.STARTING, base.BatchStatus.IN_PROGRESS):
+            for future in self._futures:
+                future.result()
+
```

Comment:
Timeout should be added maybe!?

Reply:
I would prefer adding a `timeout` argument.